debugger/src/debug_adapter/dap_adapter.rs

```diff
@@ -1244,10 +1230,80 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
         // The EXACT number of instructions to return in the result.
         instruction_count: i64,
     ) -> Result<Vec<dap_types::DisassembledInstruction>, DebuggerError> {
-        let instruction_offset_as_bytes = (instruction_offset
-            * target_core.core_data.debug_info.get_instruction_size() as i64)
-            / 4
-            * 5;
+        let cs = match target_core.core.instruction_set()? {
+            InstructionSet::Thumb2 => {
+                let mut capstone_builder = Capstone::new()
+                    .arm()
+                    .mode(armArchMode::Thumb)
+                    .endian(Endian::Little);
+                if matches!(target_core.core.core_type(), CoreType::Armv8m) {
+                    capstone_builder = capstone_builder
+                        .extra_mode(std::iter::once(capstone::arch::arm::ArchExtraMode::V8));
+                }
+                capstone_builder.build()
+            }
+            InstructionSet::A32 => {
+                // TODO: We need to inspect the CPSR to determine what mode this is opearting in
+                let mut capstone_builder = Capstone::new()
+                    .arm()
+                    .mode(armArchMode::Arm)
+                    .endian(Endian::Little);
+                if matches!(target_core.core.core_type(), CoreType::Armv8m) {
```

Comment:
This whole if block isn't needed - no Cortex-M supports the A32 instruction set.  They're all Thumb2 only.

Reply:
Thanks :)

Reply:
What about the V8 part of the logic?

Reply:
I believe it's only needed for Thumb2, where there is a v8 extension.

For the "full ARM" ISA A32 is v7-a, A64 is v8-a, and when a v8 chip is speaking A32 it's in a backwards compatibility mode with no extensions.

Reply:
Thanks :)